java/kotlin-extractor2/src/main/kotlin/entities/Expression.kt

```diff
@@ -215,25 +221,57 @@ OLD: KE1
         }
 */
 
-private fun isFunction(
+private fun isFunctionMatchingNames(
+    symbol: KaFunctionSymbol,
+    name: CallableId,
+    isExtension: Boolean = false,
+    nullability: KaTypeNullability = KaTypeNullability.UNKNOWN,
+    extensionReceiverClassName: ClassId? = null
+): Boolean {
+
+    if (symbol.callableId != name)
+        return false
+
+    if (!isExtension)
+        return true
```

Comment:
I'm a bit confused what's going on here. Should this be `if (!isExtension && receiverType == null)`?

Reply:
I reworked this a bit in an extra commit. My goal was to only use the last two parameters if `isExtension==true`.